submariner/globalnet-enhancement2-0.md

```diff
@@ -0,0 +1,298 @@
+    Status GlobalEgressIPStatus `json:"status,omitempty"`
+
+    // +optional
+    Message *string `json:"message,omitempty"`
+
+    // The list of GlobalIPs assigned via this GlobalnetEgressIP object.
+    GlobalIPs []string `json:"globalIPs"`
+}
+```
+
+When Submariner Globalnet is deployed in a Cluster, it auto-allocates the configured number of
+`globalIPs` and programs the necessary egress rules on the active Gateway node of the Cluster.
+This is done internally by creating `ClusterGlobalEgressIP` CR object with the name `cluster-default`.
+Users can query the CR to check the allocated GlobalIPs at the Cluster level. If users create
+multiple ClusterGlobalEgressIP CRs that match the Cluster scope, one of the egressIP at the Cluster
+scope will be used.
+
+Only an admin user can create the `ClusterGlobalEgressIP` object and regular users will only be able
+to create `GlobalEgressIP` objects.
+
+### Backward Compatibility
+
+Inorder to support the newer implementation, a combination of ipSets with multiple IPTable chains
+will be used. Also, we will internally use configMaps to maintain the state of allocated GlobalIPs
+and get rid of annotations on the objects. Because of this, backward compatibility cannot be maintained
+with the existing implementation of Globalnet.
+
+### User Stories
+
+1.User has two (or more) clusters with Overlapping CIDRs and requires the Pods in one
+  Cluster be able to discover and talk to exported Services.
+
+  To achieve this use-case, user can simply deploy Submariner Globalnet on all the participating
+  clusters. By default, a globalIP (or a set of globalIPs) will be assigned at the Cluster level
+  which will be used as egressIPs for inter-cluster communication and necessary Services can be
+  exported.
+
+2.In a Globalnet deployment user wants all the Pods in a namespace `ns1` to use a unique globalIP
+  instead of the globalIP assigned at the cluster level.
+
+  To achieve this, one can apply the following CRD in the namespace `ns1`.
+
+```yaml
+   apiVersion: submariner.io/v1alpha1
+   kind: GlobalEgressIP
+   metadata:
+     name: ns-egressip
+     namespace: ns1
+   spec:
+     numGlobalIPs: 1
+```
+
+3.In a Globalnet deployment, user wants a selected Pod (or set of Pods) in a particular namespace
+  to use a unique globalIPs as egressIP for cross-cluster communication.
+
+  To achieve this, one can apply the following CRD in the namespace `ns1`.
+
+```yaml
+   apiVersion: submariner.io/v1alpha1
+   kind: GlobalEgressIP
+   metadata:
+     name: db-pods
+     namespace: ns1
+   spec:
+     podSelector:
+         matchLabels:
+           role: db
+     NumGlobalIPs: 2
+```
+
+5.In a Globalnet deployment, user wants a specific service to be made available to remote clusters.
+  For this, user can create a ClusterIP Service and export the Service. Globalnet will allocate a
+  globalIP for ingress connectivity such that any external applications can connect to this Service.
```

Comment:
This seems a duplicate of 1.

Reply:
1 talks about EgressIP support whereas this is about IngressIP support.